### ISSUE TITLE
[ENH] Introduce tests to validate conformity of DL estimators

### ISSUE THREAD

**Issue by achieveordie** [sktime/sktime#4535]
**Is your feature request related to a problem? Please describe.**
DL estimators are usually based on certain assumptions (save/load is more comprehensive, expected input dimension is [transposed](https://github.com/sktime/sktime/pull/4531#issuecomment-1529740511) that of normal estimators, expect private attributes like `network_`, `model_` etc) to properly function. This creates multiple points of failure and missing even one of these things creates a buggy estimator could need several PRs to test and debug.

Hence, I propose adding a test that checks for this conformity to avoid situations like #4531, #4527.

**Describe the solution you'd like**
Tests that run only on child classes of `BaseDeepRegressor` and `BaseDeepClassifier` (and future Base-Deep Abstract Estimators) to check if the newly added estimator completely conforms to the expected design.

**Describe alternatives you've considered**
An alternative solution to this problem would involve a lot of human intervention - writing down steps to adding new DL estimators, reviewers checking off each point mentioned in the fore-mentioned recipe, or something like that. 

**Additional context**

**Comment by fkiraly** [comment]
Agreed! I think we also want to write up the extension contract somewhere.
The RNN estimator has caused me so much headache, running after these undocumented extension assumptions one by one...

Further, from a design and architecture perspective, I think we should also think carefully about what we can do to simplify the contract. A contract that adds only a minimum above the general classifier contract would be best, so what is the minimum we can get away with?

**Comment by fkiraly** [comment]
Hm, my gut feeling agrees with yours - let's write the tests first, and then we can see whether we can bootstrap docstrings or similar into a full contract specification.
